packages/core/src/pagination/pagination.element.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { baseStyles, I18nService, property } from '@cds/core/internal';
+import { html, LitElement } from 'lit-element';
+import { cache } from 'lit-html/directives/cache.js';
+import { CdsInput } from '../input/input.element.js';
+import { styles } from './pagination.element.css.js';
+
+export const PAGINATION_LAYOUT = 'horizontal gap:md align:center';
+
+/**
+ * Web component pagination.
+ *
+ * ```typescript
+ * import '@cds/core/pagination/register.js';
+ * ```
+ *
+ * ```html
+ * <cds-pagination>
+ *   <cds-pagination-button action="first" disabled></cds-pagination-button>
+ *   <cds-pagination-button action="prev" disabled></cds-pagination-button>
+ *   <span aria-label="current page">1 / 3</span>
+ *   <cds-pagination-button action="next"></cds-pagination-button>
+ *   <cds-pagination-button action="last"></cds-pagination-button>
+ * </cds-pagination>
+ * ```
+ *
+ * @element cds-pagination
+ * @slot
+ * @slot cds-pagination-button
+ * @cssprop --color
+ * @cssprop --font-size
+ */
+
+export class CdsPagination extends LitElement {
+  /**
+   * @default pagination
+   */
+  @property({ type: String, reflect: true }) ariaLabel = I18nService.keys.pagination.paginationAriaLabel;
+
+  render() {
+    const userLayout = html`<slot></slot>`;
+    const defaultLayout = html`<div cds-layout="${PAGINATION_LAYOUT}"><slot></slot></div>`;
+
+    return html`${cache(this.hasAttribute('cds-layout') ? userLayout : defaultLayout)}`;
+  }
+
+  static get styles() {
+    return [baseStyles, styles];
+  }
+
+  firstUpdated() {
+    this.setupNumberInput();
+  }
+
+  updated(props: Map<string, any>) {
+    super.updated(props);
+  }
+
+  private setupNumberInput() {
+    const numberInput = this.querySelector<CdsInput>('[cds-pagination-number]');
+    if (numberInput) {
+      numberInput.controlWidth = 'shrink';
+      numberInput.layout = 'compact';
+    }
+  }
```

Comment:
To be consistent with the rest of the library let's use `@query` (import from 'lit-element'):

You can mark it as private so it won't show up on the dox:

```
export class CdsPagination extends LitElement {
  ...
  
  /** @private */
  @query('cds-input[cds-pagination-number]') numberInput: CdsInput;
  
  ...

  private setupNumberInput() {
      if (this.numberInput) {
        this.numberInput.controlWidth = 'shrink';
        this.numberInput.layout = 'compact';
      }
    }
  ```